include/ginkgo/core/base/executor.hpp

```diff
@@ -724,6 +725,20 @@ class Executor : public log::EnableLogging<Executor> {
 
 #undef GKO_ENABLE_VERIFY_MEMORY_TO
 
+    /**
+     * Populates the executor specific info from the global machine topology
+     * object.
+     *
+     * @param mach_topo the machine topology object.
+     */
+    virtual void populate_exec_info(const MachineTopology *mach_topo) = 0;
+
+    struct exec_info {
+        int device_id = -1;
+        int num_cores = 0;
+        int num_work_groups_per_core = 0;
```

Comment:
Does `num_pu_per_core` make sense for GPU ? Isn't the `work_group` kind of equivalent to a `pu` ? I think `pu` does not really make sense for a GPU

Reply:
A workgroup is an expression of a unit of computation with internal parallelism which a single computation unit (i.e. core or SM) is able to execute as a standalone. In OpenCL, a computation unit pulls a workgroup (a bunch of workitems/logical threads) which it will execute.
In OpenCL, subgroup would be closer to PU.

Then, how about something like `subpu_per_core` or something like that. I think we agree that core represents both a CPU core and a SM fairly well, then what we need is a generic enough term to represent a sub processing unit of a core/SM, hence simply subPU, but maybe something else has a better idea.